misc_test.go

```diff
@@ -294,47 +295,37 @@ func TestNaturalLess(t *testing.T) {
 	}
 }
 
+type fakeFileInfo struct {
+	name  string
+	isDir bool
+}
+
+func (fileinfo fakeFileInfo) Name() string       { return fileinfo.name }
+func (fileinfo fakeFileInfo) Size() int64        { return 0 }
+func (fileinfo fakeFileInfo) Mode() os.FileMode  { return os.FileMode(0o000) }
+func (fileinfo fakeFileInfo) ModTime() time.Time { return time.Unix(0, 0) }
+func (fileinfo fakeFileInfo) IsDir() bool        { return fileinfo.isDir }
+func (fileinfo fakeFileInfo) Sys() any           { return nil }
+
 func TestGetFileExtension(t *testing.T) {
-	dir := t.TempDir()
 	tests := []struct {
 		name              string
 		fileName          string
+		isDir             bool
 		expectedExtension string
 	}{
-		{"normal file", "file.txt", ".txt"},
-		{"file without extension", "file", ""},
-		{"hidden file", ".gitignore", ""},
-		{"hidden file with extension", ".file.txt", ".txt"},
+		{"normal file", "file.txt", false, ".txt"},
+		{"file without extension", "file", false, ""},
+		{"hidden file", ".gitignore", false, ""},
+		{"hidden file with extension", ".file.txt", false, ".txt"},
+		{"directory", "dir", true, ""},
```

Comment:
Thanks, I think this looks better now. Perhaps you can add more test cases to reflect the actual purpose of this PR:

```go
{"hidden directory", ".git", true, ""},
{"directory with dot", "profile.d", true, ""},
```